esp_flasher/__main__.py

```
@@ -241,11 +241,74 @@
+
+    
+    if sys.platform == 'win32':
+        import ctypes
+        params = " ".join(f'"{arg}"' for arg in args[1:])
+        ctypes.windll.shell32.ShellExecuteW(
+            None, "runas", python_exe, params, None, 1
+        )
+        sys.exit(0)
+
+    elif sys.platform == 'darwin':
+        cmd = f'do shell script "cd \\"{cwd}\\"; {python_exe} {" ".join(args[1:])}" with administrator privileges'
```

Ruff (0.12.2)

Undefined name `cwd`

(F821)